apps/hexdump.c

```
@@ -17,49 +18,52 @@ static void print_hex(unsigned int val, int width)
         fclose(fp);
-        return 1;
+    } else {
+        int cap = 512;
+        data = malloc(cap);
+        if (!data) { print("hexdump: out of memory\n"); return 1; }
+        char buf[512];
+        int n;
+        while ((n = sys_read_fd(0, buf, sizeof(buf))) > 0) {
+            while (total + n > cap) {
+                cap *= 2;
+                data = realloc(data, cap);
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)